all.tf

```
@@ -0,0 +1,100 @@
+  statement {
+    sid    = "DatadogAll"
+    effect = "Allow"
+
+    actions = [
+        "apigateway:GET",
+        "autoscaling:Describe*",
+        "budgets:ViewBudget",
+        "cloudfront:GetDistributionConfig",
+        "cloudfront:ListDistributions",
+        "cloudtrail:DescribeTrails",
+        "cloudtrail:GetTrailStatus",
+        "cloudwatch:Describe*",
+        "cloudwatch:Get*",
+        "cloudwatch:List*",
+        "codedeploy:List*",
+        "codedeploy:BatchGet*",
+        "directconnect:Describe*",
+        "dynamodb:List*",
+        "dynamodb:Describe*",
+        "ec2:Describe*",
+        "ecs:Describe*",
+        "ecs:List*",
+        "elasticache:Describe*",
+        "elasticache:List*",
+        "elasticfilesystem:DescribeFileSystems",
+        "elasticfilesystem:DescribeTags",
+        "elasticloadbalancing:Describe*",
+        "elasticmapreduce:List*",
+        "elasticmapreduce:Describe*",
+        "es:ListTags",
+        "es:ListDomainNames",
+        "es:DescribeElasticsearchDomains",
+        "health:DescribeEvents",
+        "health:DescribeEventDetails",
+        "health:DescribeAffectedEntities",
+        "kinesis:List*",
+        "kinesis:Describe*",
+        "lambda:AddPermission",
+        "lambda:GetPolicy",
+        "lambda:List*",
+        "lambda:RemovePermission",
+        "logs:Get*",
+        "logs:Describe*",
+        "logs:FilterLogEvents",
+        "logs:TestMetricFilter",
+        "logs:PutSubscriptionFilter",
+        "logs:DeleteSubscriptionFilter",
+        "logs:DescribeSubscriptionFilters",
+        "rds:Describe*",
+        "rds:List*",
+        "redshift:DescribeClusters",
+        "redshift:DescribeLoggingStatus",
+        "route53:List*",
+        "s3:GetBucketLogging",
+        "s3:GetBucketLocation",
+        "s3:GetBucketNotification",
+        "s3:GetBucketTagging",
+        "s3:ListAllMyBuckets",
+        "s3:PutBucketNotification",
+        "ses:Get*",
+        "sns:List*",
+        "sns:Publish",
+        "sqs:ListQueues",
+        "support:*",
+        "tag:GetResources",
+        "tag:GetTagKeys",
+        "tag:GetTagValues",
+        "xray:BatchGetTraces",
+        "xray:GetTraceSummaries"
+    ]
+
+    resources = ["*"]
+  }
+}
+
+module "all_label" {
+  source     = "git::https://github.com/cloudposse/terraform-null-label.git?ref=0.2.1"
+  namespace  = "${var.namespace}"
+  stage      = "${var.stage}"
+  name       = "${var.name}"
+  attributes = ["${compact(concat(var.attributes, list("all")))}"]
+}
+
+locals {
+  all_count = "${contains(split(",", lower(join(",", var.integrations))), "all") ? 1 : 0}"
+}
+
+resource "aws_iam_policy" "all" {
+  count  = "${local.all_count}"
+  name   = "${module.all_label.id}"
+  policy = "${data.aws_iam_policy_document.all.json}"
+}
+
+resource "aws_iam_role_policy_attachment" "all" {
+  count      = "${local.all_count}"
+  role       = "${aws_iam_role.default.name}"
+  policy_arn = "${aws_iam_policy.all.arn}"
```

```suggestion
  policy_arn = "${join("", aws_iam_policy.all.*.arn)}"
```

Use `splat+join` for all resources with `count`.
If `count` is added, the resource becomes a list (even with a single item).
To get an item from the list, use `splat`.
To prevent Terraform from complaining, use `join` with an empty string - in this case when `count=0`, the `splat+join` expression will produce an empty string.